### PR TITLE
Add GitHub Actions for Docker

### PR DESCRIPTION
**Pull request by ulrikstrid**
## Problem

We want the docker image to be built automatically

## Solution

Adds github actions and pushes to the github registry (for now, we can add Docker hub later)

## Related

- #337

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Build and push
+        # Only push if on main branch
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          file: ./docker/alpine.Dockerfile
+          context: .
+          platforms: linux/amd64,linux/arm64
+          push: github.ref == 'refs/head/main' && github.event_name == 'push'
+          tags: marigold-dev/deku:latest
```
Comment (callistonianembrace) [inline .github/workflows/docker.yml:38]: We need commit hash in the tags to trace things when deployments need debugging.

Comment (callistonianembrace) [reply .github/workflows/docker.yml:38]: *too

Comment (callistonianembrace) [reply .github/workflows/docker.yml:38]: Not saying `latest` images are not needed. Perhaps another PR could deal with such tags

**Review by callistonianembrace — APPROVED** [review]

File: .github/workflows/docker.yml (hunk 1 above)
Comment (ulrikstrid) [reply .github/workflows/docker.yml:38]: I need to fix the boolean expression above, will add a commit hash tag as well 👌

Comment (ulrikstrid) [reply .github/workflows/docker.yml]: Will tag with `latest` and the git sha

**Pull request closed without merging**